spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/service/impl/DefaultStreamService.java

```diff
@@ -416,17 +426,45 @@ private boolean isPartitionedConsumer(Map<String, String> appDeploymentPropertie
 
 	@Override
 	public Map<StreamDefinition, DeploymentState> state(List<StreamDefinition> streamDefinitions) {
-		if (System.getProperty("USE_SKIPPER") != null) {
-			return this.skipperStreamDeployer.state(streamDefinitions);
-		} else {
-			return this.appDeployerStreamDeployer.state(streamDefinitions);
+		Map<StreamDefinition, DeploymentState> states = new HashMap<>();
+		List<StreamDefinition> skipperStreams = new ArrayList<>();
+		List<StreamDefinition> appDeployerStreams = new ArrayList<>();
+		for (StreamDefinition streamDefinition: streamDefinitions) {
+			StreamDeployment streamDeployment = this.streamDeploymentRepository.findOne(streamDefinition.getName());
+			if (streamDeployment == null) {
+				states.put(streamDefinition, DeploymentState.unknown);
+			}
+			else {
+				if (streamDeployment.getDeployerName().equals(StreamDeployers.skipper.name())) {
+					skipperStreams.add(streamDefinition);
+				}
+				else if (streamDeployment.getDeployerName().equals(StreamDeployers.appdeployer.name())){
+					appDeployerStreams.add(streamDefinition);
+				}
+				else {
+					logger.error("Unknown deployer " + streamDeployment.getDeployerName() +
```

Comment:
Just wondering if we should throw an exception here.

Reply:
Currently, we have the pre-defined deployer names and it is most likely to have this path of the code to execute. Given this is for state() calculation and throwing an exception for one incorrect entry may not be necessary. Hence, just error logging.

Reply:
Understood.   I guess the log is enough, just was concerned about leaving an entry out of the result.